[Gabriel Chia] Duke Increments

gachia opened the pull request:
iP Duke Increments

j-lum committed:
Add support for Gradle workflow

j-lum committed:
Add sample checkstyle configuration

Add toolVersion block in to Gradle code sample to prevent errors.

gachia committed:
Level-1

Completed Level 1:
Add Greet, Echo and Exit to Duke

gachia committed:
Level-2

Completed Level 2:
Add, List

gachia committed:
Level 3

Completed Level 3. Mark as Done

Implement "todo" command from level 4 for use to add tasks with multiple words

gachia committed:
Level 4

Completed Level 4: ToDos, Events, Deadlines

gachia committed:
Level 5

Completed Level 5: Handle Errors

gachia committed:
Level 6

Completed Level 6: Delete

gachia committed:
Level 7

Level 7 Save completed

gachia committed:
Level 8

Level 8: Dates and Times completed

gachia committed:
Level 8 Updated

Move some code for clean merge

gachia committed:
Merge branch 'branch-Level-8'

gachia committed:
Merge branch 'branch-Level-7'

gachia committed:
Merge branch 'master' of https://github.com/gachia/duke

gachia committed:
Level A-MoreOOP

Level A-MoreOOP completed.
Added Command and child class to execute each command.
Added Parser to handle initial command.
Added Ui to handle messages.
Added Storage to handle save and load.
Added TaskList to hold tasks.

gachia committed:
Merge branch 'gradle'

gachia committed:
Added Gradle

Added Gradle to the project. Implementation of checkstyle done partially.
Implementation of JUnit testing done partially

gachia committed:
Completed A-JUnit

Completed Level A-JUnit.

Added JUnit test cases for execute method for AddCommand, DeleteCommand and DoneCommand

gachia committed:
Completed A-JavaDoc

Completed Level A-JavaDoc

Fixed several bugs:
- Enable a line in gradle to run Duke with System input, otherwise it will not do so and fail the build upon running
- Fixed Exit Command to exit Duke when file path is not specified. Previously it did not exit due to throwing a DukeException which skipped its exit line.

gachia committed:
Merge branch 'branch-A-JavaDoc'

gachia committed:
Updated Jar version to 0.1.4

jar version updated to v0.1.4

gachia committed:
Fixed missing * symbol

Fixed missing * symbol to make my Javadocs comments to be recognisable

gachia committed:
Completed A-CodingStandard

Completed Level A-CodingStandard.

Used checkstyle to ensure code complies with standard

gachia committed:
Completed Level 9

Completed Level 9 Find

gachia committed:
Merge branch 'branch-A-CodingStandard'

gachia committed:
Merge branch 'branch-Level-9'

mrnewguy commented on src/main/java/AddCommand.java:
```diff
@@ -0,0 +1,79 @@
+
+    private String taskCmd;
+    final String timePattern = "d MMMM yyyy, h:mma";
+    DateTimeFormatter dateTimeFormat = DateTimeFormatter.ofPattern(timePattern);
```
Good use of the java DateTimeFormatter! However, I think that you could try implementing your own DateTimeFormatter class too!

mrnewguy replied:
Good use of the java DateTimeFormatter! However, I think that you could try implementing your own DateTimeFormatter class too!

mrnewguy commented on src/main/java/AddCommand.java:
```diff
@@ -0,0 +1,79 @@
+            tasks.addTask(new Event(userWords[0].trim(), userWords[1].trim()));
+            break;
+        default:
+            throw new DukeException("Wrong type of command entered" + taskCmd);
```
Nice use of switch case statements, keeps code clean. However, does this imply that the default case is always an "exception"?

gachia replied:
Yes, a default command isn't a recognisable command by Duke and should throw an exception.

mrnewguy commented on src/main/java/Ui.java:
```diff
@@ -0,0 +1,80 @@
+/**
+ * Represents the User Interface of Duke, including the user input fields.
+ */
+public class Ui {
```
In general, I think the abstractions are well done (e.g.  showError, showGoodbye). It makes the code readable and more importantly highly extensible

mrnewguy reviewed (commented):
Overall well done. Code is clean and well-abstracted. I especially like the use of switch cases and a "finally" block to throw errors. However, I am not sure whether this is the convention or whether there is a better way to do it.

gachia committed:
Cleaned code with Checkstyle

Used checkstyle to clean up coding style violations.

gachia committed:
Created GUI part 1

Followed tutorial to create a test GUI

brianyenna reviewed (commented):
Good code overall! Clear and concise. But perhaps there could be more separation of concerns. For instance, in the switch-case statement used in Duke.java, perhaps the handling of each of the individual commands could be abstracted away to a separate method call dedicated to that method. There are also some minor coding style violations (mainly pertaining to the inconsistent spacing between the curly braces and surrounding code)

brianyenna commented on src/main/java/Duke.java:
```diff
@@ -1,163 +1,48 @@
-                }
-            } catch (Exception e) {
-                System.out.println(e.getMessage());
+                String fullCommand = ui.readCommand();
```
Perhaps a possible alternative implementation is to get the Parser to parse the entire input, not just the command, and allow the Parser to decide which are the parameters to pass on to the various Command objects. Hence, the different Command objects have constructors that take in the remaining parameters (such as the date and time for the DeadlineTask). This way, there is no need to make further UI calls within the Command objects, and the Command object can instead be used solely for decision-making logic. Perhaps we can debate over the merits of either approach :)

gachia replied:
Your alternative implementation seems cleaner. Mine was a hack-ish way of getting the data part of the input since the Scanner object was inside the UI class.

brianyenna reviewed (commented):
Great code overall. Gave me some good ideas as well :)

gachia committed:
Completed Level 10

Completed Level 10 GUI
- Implemented Duke to interact with GUI
- Formatted code
- Added Javadocs for new code
- Added more OOP to AddCommand, by separating into three commands: todo, deadline, event
- Fixed test cases (+2 squashed commit)

Squashed commit:

[6a27cd6] Replaced copyright images

Replaced copyright images with non-copyright images

[aceaef8] Completed JavaFX tutorial

Completed setting up GUI from JavaFX Tutorial.
Next: Set up Duke with GUI

gachia committed:
Completed JavaFX tutorial

Completed setting up GUI from JavaFX Tutorial.
Next: Set up Duke with GUI

gachia committed:
Replaced copyright images

Replaced copyright images with non-copyright images

gachia committed:
Completed Level 10

Completed Level 10 GUI
- Implemented Duke to interact with GUI
- Formatted code
- Added Javadocs for new code
- Added more OOP to AddCommand, by separating into three commands: todo, deadline, event
- Fixed test cases

gachia committed:
Merge branch 'branch-Level-10' of https://github.com/gachia/duke into branch-Level-10

gachia committed:
Merge branch 'branch-A-Checkstyle'

gachia committed:
Merge branch 'branch-Level-10'

gachia committed:
Updated Project version

Updated project version and generated a new Jar file

gachia committed:
Fixed main class run name

Fixed the gradle main class name that prevented it for running

gachia committed:
Added Assertion

Added assertion checks in setDone and delete functions of Tasklist. Enabled assertions in gradle build

gachia committed:
Merge pull request #4 from gachia/branch-A-Assertions

Added Assertion

gachia committed:
Refactor Ui code

Abstracted Add, Delete and Done printing code to the Ui class as functions.
Also fixed assertion bug and test cases.

gachia committed:
Merge pull request #5 from gachia/branch-A-CodeQuality

Refactored Ui code

gachia committed:
Completed C-Help

Added Help Command for the user to see the list of Commands available to use.
Also aded a window icon for Duke.

gachia committed:
Merge pull request #7 from gachia/branch-C-Help

Completed C-Help

gachia committed:
Added User Guide

Added a User Guide for Duke.
Updated command messages to reflect Duke's new personality.

gachia committed:
Merge pull request #8 from gachia/branch-A-UserGuide

Added User Guide

gachia committed:
Fixed Readme

Fixed Readme bug where text is not formatted correctly

gachia committed:
Merge branch 'branch-A-UserGuide'

gachia committed:
Updated User Guide

Updated User Guide with help command guide.
Also added more personality to Duke.

gachia committed:
Added CSS style to Duke

Added CSS to make Duke look prettier.
Added background image for Duke.
Small addition to exit message.
Set window to un-resizable.
Edited User Guide to include saving upon exit.
Added References in User Guide.

gachia committed:
Added rounded corners to text box

Added rounded corners to TextArea.
Update Ui image to reflect changes.

gachia committed:
Preparing for release

- Updated dependencies to have application work on all platform
- Fixed save and load with jar file
- Moved data files into resources folder
- Updated Dialogbox fields
- Removed emoji from exception messages

gachia committed:
Remove unnecessary dependencies

remove unnecessary javafx dependencies
